File: app/src/main/java/co/medellinandroid/spotifyartists/model/SpotifyTrack.java
package co.medellinandroid.spotifyartists.model;

import com.squareup.moshi.Json;

public class SpotifyTrack {

  @Json(name = "id") private String id;
  @Json(name = "name") private String name;
  @Json(name = "duration_ms") private int duration;
  @Json(name = "album") private SpotifyAlbum album;

  public SpotifyTrack() {
    this.album = new SpotifyAlbum();
  }

  public String getId() {
    return id;
  }

  public void setId(String id) {
    this.id = id;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public int getDuration() {
    return duration;
  }

  public void setDuration(int duration) {
    this.duration = duration;
  }

  public SpotifyAlbum getAlbum() {
    return album;
  }

  public void setAlbum(SpotifyAlbum album) {
    this.album = album;
  }
}

File: app/src/main/java/co/medellinandroid/spotifyartists/model/SpotifyTrack.kt
package co.medellinandroid.spotifyartists.model

import com.squareup.moshi.Json

class SpotifyTrack {

  @Json(name = "id")
  var id: String? = null
  @Json(name = "name")
  var name: String? = null
  @Json(name = "duration_ms")
  var duration: Int = 0
  @Json(name = "album")
  var album: SpotifyAlbum? = null

  init {
    this.album = SpotifyAlbum()
  }
}
